core/src/main/java/io/methvin/watcher/OnTimeoutListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.methvin.watcher;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public abstract class OnTimeoutListener implements DirectoryChangeListener {
+
+  private final Object lock = new Object() {};
+
+  private final int timeout;
+
+  private ScheduledExecutorService service;
+  private ScheduledFuture currentTask;
+
+  public OnTimeoutListener(int timeout) {
```

Comment:
just for clarity I'd name it with the units: `timeoutMillis`. We could also have a length and a `TimeUnit` but not sure if that's overkill.

Reply:
Also maybe we should just throw an `IllegalArgumentException` if the timeout is negative? Is there a use case for using this listener without a timeout?

Reply:
@gmethvin nope, i am pretty sure nope